GWAS/LMM.ipynb

```diff
@@ -1208,6 +1209,156 @@
     "1. A file with association results for each chromosome (Note: this are given in regard to Allele 2)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "kernel": "SoS"
+   },
+   "source": [
+    "## GMMAT workflow implementation\n",
+    "Documentation can be found [here](https://github.com/hanchenphd/GMMAT/blob/master/inst/doc/GMMAT.pdf)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "kernel": "SoS"
+   },
+   "source": [
+    "### Step1: Creation of the GRM "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "kernel": "SoS"
+   },
+   "outputs": [],
+   "source": [
+    "#Calculate standardized GRM using GEMMA\n",
+    "[gemma_grm]\n",
+    "input:bfile\n",
+    "output:f'{cwd}/{bfile:bn}.sXX.txt'\n",
+    "task: trunk_workers = 1, trunk_size = job_size, walltime = '2h', mem = '6G', cores = 1, tags = f'{step_name}_{_output[0]:bn}'\n",
+    "bash: container=container_lmm, expand = \"${ }\", stderr = f'{_output:n}.stderr', stdout = f'{_output:n}.stdout' \n",
+    "    gemma \\\n",
+    "    -bfile ${_input:n} \\\n",
+    "    -gk 2 \\\n",
+    "    -o ${_output:bnn}\n",
+    "bash: container=container_lmm,expand='${ }', stderr = f'{_output[0]:n}.stderr', stdout = f'{_output[0]:n}.stdout'\n",
+    "    cp ./output/${bfile:bn}.sXX.txt ${cwd}\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "kernel": "SoS"
+   },
+   "source": [
+    "### Step 2 :Fitting the null"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "kernel": "SoS"
+   },
+   "outputs": [],
+   "source": [
+    "# Run GMMAT step1: fit a GLMM with covariate adjustment and random effects to account for population structure and family or cryptic relatedness\n",
+    "[GMMAT_1]\n",
```

Comment:
Because your current GMMAT_2 step takes other inputs and not the first step, I suggest you rename `GMMAT_1` to `GMMAT_null` for the null model, then rename `GMMAT_2` to `GMMAT`. When you run `GMMAT` it should trigger GMMAT_null, which will then trigger `gemma_grm`. This is the feature of a workflow language.